src/Components/StepThree/StepThree.js

```diff
@@ -1,112 +1,294 @@
+            ...prevState,
+            [e.target.name]: !!e.target.value
+                }))
+    };
+
+    const handleChangeConvictions = (e, i) => {
+        const {name, value} = e.target;
+        const list = [...inputs.convictions];
+        list[i][name] = value;
+        setInputs((prevState) => ({
+            ...prevState,
+            convictions: list,
+        }))
+    }
+
+    const handleRemoveClick = (i) => {
+        const list = [...inputs.convictions];
+        list.splice(i, 1);
+        setInputs((prevState) => ({
+            ...prevState,
+            convictions: list,
+        }))
+    }
+
+    const handleAddClick = () => {
+        setInputs((prevState) => ({
+            ...prevState,
+            convictions: [...inputs.convictions, {forWhat: '', convictionDate: ""}]}))
+    }
+
+    const handleCheck = (e) => {
+        let newArray = [...inputs.metalworkSkills, e.target.value];
+        let {value} = e.target;
+        if (inputs.metalworkSkills.includes(value)) {
+            newArray = newArray.filter((elem) => elem !== value)
+        }
+        setInputs((prevState)=>({
+            ...prevState,
+            metalworkSkills: newArray,
+        }))
+    }
+    useEffect(() => {
+        const stringMetalWorkSkills = inputs.metalworkSkills.toString();
+          setData((data) => ({
+            ...data,
+            doesHaveAgricultureSkills: inputs.doesHaveAgricultureSkills,
+            agricultureSkills: inputs.agricultureSkills,
+            doesHaveMetalworkSkills: inputs.doesHaveMetalworkSkills,
+            metalworkSkills: stringMetalWorkSkills,
+            isConvicted: inputs.isConvicted,
+            convictions: inputs.convictions,
+            doesFlyAirplane: inputs.doesFlyAirplane,
+            doesDriveCar: inputs.doesDriveCar,
+            doesDriveBicycle: inputs.doesDriveBicycle,
+          }));
+        
+      }, [inputs]);
 
     return(
       <div className="wizardContainer">
           <div className="inputContainer">
               <div className="inputHeader">
                   <h4>Step 3</h4>
-                  <p>Mandatory fields are labeld with *</p>
+                  <p>Mandatory fields are labeled with *</p>
               </div>
           </div>
 
         <div className="thirdInputFields">
 
-           <div className="skills">
+        <div className="skills">
             <p><span>*</span> Do you have any agriculture skills?</p>
             <div>
                 <div>
-                <input type="radio" id="skillsAgrYes" value="yes" name="skAgr"/>
-                <label for="skillsAgrYes">Yes</label>
+                <Input 
+                        type={"radio"}
+                        id={"doesHaveAgricultureSkillsYes"}
+                        value={true} 
+                        name={"doesHaveAgricultureSkills"} 
+                        onChange={handleChangeRadio}/>
+                <label htmlFor="doesHaveAgricultureSkillsYes">Yes</label>
```

Comment:
Please be consistent with props values - if the value is a string, there is no need to add brackets (should be like in label: htmlFor="doesHaveAgricultureSkillsYes"). Please apply this change across all codebase.